app/src/main/java/org/kiwix/kiwixmobile/nav/destination/reader/KiwixReaderFragment.kt

```diff
@@ -311,6 +307,28 @@ class KiwixReaderFragment : CoreReaderFragment() {
     return ShouldCall
   }
 
+  private fun getZimFileFromUri(
+    uri: Uri
+  ): File? {
+    val filePath = FileUtils.getLocalFilePathByUri(
+      requireActivity().applicationContext, uri
```

Comment:
@MohitMaliDeveloper we should add tests for this. Maybe in a new PR. This PR is too old now.

Reply:
Okay sure.

Reply:
@MohitMaliDeveloper In such case, please update immediatly a new issue, I think this has not been done.